### PR TITLE
Add initial baremetal IPI docs

### LINKED ISSUE #67: Address documentation gaps
https://github.com/openshift-metal3/kni-installer/blob/master/docs/dev/baremetal.md

Our dev doc includes some information that's only relevant to the fork.  To prepare a PR, should move the fork-specific content elsewhere.  Perhaps we could use a wiki page on the kni-installer repo?  Some of the dev doc content makes sense to keep in the repo.

We also need to create a user doc that covers how the platform is used, prerequisites, install config syntax, and so forth.

### PR DESCRIPTION
**Pull request by stbenjam**
fixes #67

In order to get an experimental version of the baremetal platform in openshift/installer, we need to have some basic docs about how to use it.

The documentation about our fork, while it exists, can be moved into a wiki on this repo.  This PR restores the openshift readme, and adds some basic docs about the platform.

### CODE REVIEW

File: docs/user/metal/install_ipi.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Install: Baremetal Installer-Provided Infrastructure (IPI)
+
+TBD.
```
Comment (stbenjam) [inline docs/user/metal/install_ipi.md:3]: I figure it's easier to collaborate on this in Google docs - will convert to markdown once we're happy w/ it.

https://docs.google.com/document/d/1r4MzBm2J_vMU42g6LDiLtCdfMI_2sgf7A7HbFVOTz3w/edit?usp=sharing

Comment (markmc) [reply docs/user/metal/install_ipi.md:3]: Looks like there hasn't been ongoing collaboration on this - I think what you have is good enough to merge and include in the next upstream PR refresh

Comment (russellb) [reply docs/user/metal/install_ipi.md:3]: It looks like the README fixes already went in, and most of the rest is in a google doc.  I wanted to work more on user docs today.  I'll incorporate the contents of the google doc and get a new PR opened later today.

**Comment by russellb** [comment]
will post a new docs PR shortly, incorporating what was started here

**Pull request closed without merging**